pravega-client-examples/src/main/java/io/pravega/example/secure/SecureReader.java

```diff
@@ -0,0 +1,96 @@
+ *
+ *    a) SSL/TLS (HTTPS) communications with a Pravega cluster for data-in-transit encryption and
+ *       server authentication.
+ *    b) Passing credentials to a Pravega cluster for authentication and authorization of clients.
+ *
+ * This example can be driven interactively against a running Pravega cluster configured to communicate using SSL/TLS
+ * and "auth" (authentication and authorization) turned on.
+ */
+public class SecureReader {
+
+    public static void main(String[] args) throws ReinitializationRequiredException {
+
+        /**
+         * Note about setting the client config for HTTPS:
+         *    - The client config below is configured to use an optional truststore. The truststore is expected to be
+         *      the certificate of the certification authority (CA) that was used to sign the server certificates.
+         *      If this is null or empty, the default JVM trust store is used. In this demo, we use a provided
+         *      "cert.pem" as the CA certificate, which is also provided on the server-side. If the cluster uses a
+         *      different CA (which it should), use that CA's certificate as the truststore instead.
+         *
+         *    - Also, the client config below disables host name verification. If the cluster's server certificates
+         *      have host names specified as the that of the server, you may turn this on. In a production
+         *      deployment, it is recommended to keep this on.
+         *
+         * Note about setting the client config for auth:
+         *    - The client config below is configured with an object of DefaultCredentials class. The user name
+         *      and password arguments passed to the object represent the credentials used for authentication
+         *      and authorization. The assumption we are making here is that the username is valid on the server,
+         *      the password is correct and the username has all the permissions necessary for performing the
+         *      subsequent operations.
+         */
+        ClientConfig clientConfig = ClientConfig.builder()
+                .controllerURI(Constants.CONTROLLER_URI) // "tls://localhost:9090"
+                .trustStore(Constants.TRUSTSTORE_PATH)   // SSL-related client-side configuration
+                .validateHostName(false)                 // SSL-related client-side configuration
+                .credentials(new DefaultCredentials("1111_aaaa", "admin")) // Auth-related client-side configuration
+                .build();
+        System.out.println("Done creating a client config.");
+
+        // Everything below depicts the usual flow of reading events. All client-side security configuration is
+        // done through the ClientConfig object as shown above.
+
+        ClientFactory clientFactory = null;
+        ReaderGroupManager readerGroupManager = null;
+        EventStreamReader<String> reader = null;
+        try {
```

Comment:
As I mentioned in notes in the PR description, I consciously avoided using `try-with-resources` to close resources automatically in order to avoid deep nesting and to ensure that most of the code reads left-justified. That is to make the code more readable. What do you say?

Reply:
Well, it's true that this nests all the code into the try block, but we used that in many other samples and it is not dramatic. Anyway, feel free to leave it as it is or even use the `@Cleanup` lombok annotation for closing resources (the annotation is nice, but it may be less obvious for a new user looking at the code).

Reply:
I left it as is for readability and easier understanding. I did not use Lombok `@cleanup` as it may not be known/understood by application developers.